bin/package-and-publish-helm-chart.py

```diff
@@ -0,0 +1,148 @@
+import os
+import re
+import requests
+import subprocess
+import sys
+
+from shutil import copyfile
+
+
+def main():
+  args = sys.argv[1:]
+  chart_path = args[0]
+  working_directory = args[1]
+  build_number = args[2]
+  is_prerelease = args[3]
+  application_version = args[4]
+
+  chart_filename = chart_path + '/Chart.yaml'
+
+  print(f'build_number = {build_number}')
+  print(f'is_prerelease = {is_prerelease}')
+  print(f'application_version = {application_version}')
+
+  username = os.getenv('ARTIFACTORY_USER')
+  token = os.getenv('ARTIFACTORY_TOKEN')
+
+  # Parse chart name from the Chart.yaml
+  chart_name = None
+  with open(chart_filename) as file:
+    for line in file:
+      if line.startswith('name:'):
+        chart_name = line.split(':')[1].strip()
+        print(f'Chart name: {chart_name}')
+        break
+
+  # Parse version from the Chart.yaml
+  chart_version = '0.0.0'
+  with open(chart_filename) as file:
+    for line in file:
+      if line.startswith('version:'):
+        chart_version = line.split(':')[1].strip()
+        print(f'Chart version: {chart_version}')
+        break
+
+  version_elements = chart_version.split('.')
+  version_major = int(version_elements[0])
+  version_minor = int(version_elements[1])
+  version_patch = int(version_elements[2])
+
+  # Retrieve the list of published helm chart versions from Artifactory
+  auth = (username, token)
+  response = requests.get(
+    url=f'https://niartifacts.jfrog.io/artifactory/rnd-helm-ci/ni/systemlink/{chart_name}/{version_major}/{version_minor}',
+    auth=auth)
+
+  # Find the highest version of the chart on Artifactory
+  highest_version = [version_major, version_minor, version_patch]
+  if response.status_code == 200:
+    print('Found existing charts on the repo...')
+    # Parse HTTP response
+    for line in response.text.splitlines():
+      if line.startswith('<a href='):
+        start = line.find('">') + len('">')
+        end = line.find('</a>')
+        package_name = line[start:end]
+        print(f'Package name: {package_name}')
+         # We need to parse the following:
+        #  chartname-0.1.27.tgz
+        #  chartname-0.1.27.tgz.prov
+        #  chartname-0.1.27-pre.20220629.4.tgz
+        pattern = r"(.+)-(\d+\.\d+\.\d+)(-pre\.\d+\.\d+)?\.tgz\S*$"
+        result = re.match(pattern, package_name)
+        groups = result.groups()
+
+        package_version = groups[1]
+        print(f'Package version: {package_version}')
+        package_version_elements = package_version.split('.')
+        package_major = int(package_version_elements[0])
+        package_minor = int(package_version_elements[1])
+        package_patch = int(package_version_elements[2])
+
+        if ((package_major >= highest_version[0]) and (package_minor >= highest_version[1]) and (package_patch > highest_version[2])):
```

Comment:
not sure if this is intended .. that seems to only work to compare 5.5.5 vs 4.4.4 but won't work when comparing 5.4.1 vs 4.4.4? 
One easy way to compare might be to stringify it with 0 padding, something like 5.4.1 becomes 050401, which is higher than 040404 ..
but now it becomes a little more involved, you might want to create a class Version to be a bit cleaner. 
Not sure if you can just use [niversion](https://dev.azure.com/ni/DevCentral/_git/Tools?path=/shared/niversion/niversion/niversion.py).BaseVersion

Reply:
Alternatively I think you can use the very lightweight `packaging` library (but you will have to add a `pip install packaging` before using it)

```python
>>> from packaging import version
>>> version.parse("5.4.1") > version.parse("4.4.4")
True
```

Reply:
This was a good catch. I'm working on getting a fix submitted to the AzDo version of this script and then I will push an update here.